fetch/fetch.go

```diff
@@ -388,15 +388,21 @@ func (f *Fetch) FetchRequestHandler(ctx context.Context, data []byte) []byte {
 		log.String("batchHash", resBatch.ID.ShortString()),
 		log.Int("numResponse", len(resBatch.Responses)),
 		log.Int("dataSize", len(bts)))
-	return bts
+	return server.SerializeResponse(f.log, bts, nil)
 }
 
 // receive Data from message server and call response handlers accordingly
-func (f *Fetch) receiveResponse(data []byte) {
+func (f *Fetch) receiveResponse(resp server.Response) {
 	if f.stopped() {
 		return
 	}
+	peerErr := resp.GetError()
+	if peerErr != nil {
+		f.log.With().Warning("received fetch error from peer", log.Err(peerErr))
+		return
```

Comment:
if returned here, this means that the request won't be removed from `pendingRequests`, is this intentional? what should happen in such case? perhaps  the error should be propagated to `req.returnChan` ?

Reply:
as @dshulyak suggested to handle error in the error handler, this is now automatically handled correctly.